crates/nu-utils/src/nu_cow.rs

```diff
@@ -0,0 +1,93 @@
+use std::fmt::Debug;
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize)]
+#[serde(untagged)]
+#[allow(dead_code)]
+pub enum NuCow<B, O> {
+    Borrowed(B),
+    Owned(O),
+}
```

Comment:
Minor thought if it matters or not. But maybe we could skip the need to import `serde` with the derive feature  for `nu-utils`` and implement the `Serialize` directly which should be pretty trivial as it dispatches to the `Serialize` of the underlying `B` and `O`. But I don't think it should matter in the compile graph that much.